cmd/kube-apiserver/app/options/customflags.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/spf13/pflag"
+
+	// ensure libs have a chance to globally register their flags
+	_ "k8s.io/apiserver/pkg/admission"
+	_ "k8s.io/kubernetes/pkg/cloudprovider/providers"
+)
+
+// AddGenericFlags explicitly registers flags that internal packages register
+// against the global flagsets from "flag". We do this in order to prevent
+// unwanted flags from leaking into the kube-apiserver's flagset.
+func AddGenericFlags(fs *pflag.FlagSet) {
+	addCloudProviderFlags(fs)
+	addAdmissionFlags(fs)
+}
+
+// addCloudProviderFlags adds flags from k8s.io/kubernetes/pkg/cloudprovider/providers.
+func addCloudProviderFlags(fs *pflag.FlagSet) {
+	// lookup flags in global flag set and re-register the values with our flagset
+	global := flag.CommandLine
+	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+
+	register(global, local, "cloud-provider-gce-lb-src-cidrs")
+
+	fs.AddFlagSet(local)
+}
+
+// addAdmissionFlags adds flags from k8s.io/apiserver/pkg/admission.
+func addAdmissionFlags(fs *pflag.FlagSet) {
+	// lookup flags in global flag set and re-register the values with our flagset
+	global := flag.CommandLine
+	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
+
+	register(global, local, "default-not-ready-toleration-seconds")
+	register(global, local, "default-unreachable-toleration-seconds")
+
+	fs.AddFlagSet(local)
+}
+
+// register adds a flag to local that targets the Value associated with the Flag named globalName in global
+func register(global *flag.FlagSet, local *pflag.FlagSet, globalName string) {
+	if f := global.Lookup(globalName); f != nil {
+		local.AddGoFlag(f)
```

Comment:
why is the logic different from https://github.com/kubernetes/kubernetes/pull/68054/commits/a77652e9ab25382706e98e24f6026e28a9fbe621#diff-2fd8f885c4e1d904b7d678e5991aade4R68 ?

Reply:
@sttts @imjching @stewart-yu we need both to match.

Reply:
@sttts @dims Originally I removed the `normalize` part because we don't need it. It was there to ensure that all globally registered flags that use underscores will be normalized to use hyphens instead. 

`klog` registers flags using underscores:
https://github.com/dims/klog/blob/master/klog.go#L411-L419

Those cloud provider and admission packages register flags using hyphens instead. I have updated the code and added the `normalize` part back. Do we actually need both files to match? The commit that was provided above was for `klog` flags and therefore we need to normalize those flags.

Also, from documentation, the two snippets of code below are equivalent:

```go
pflagFlag := pflag.PFlagFromGoFlag(f)
local.AddFlag(pflagFlag)
```

```go
local.AddGoFlag(f)
```

Reply:
seems below enough in here @dims 
```
f.Name = normalize(f.Name)
local.AddFlag(f)
```